### ISSUE TITLE
Use custom database tables for fields value store

### ISSUE THREAD

**Issue by Mte90** [CMB2/CMB2#1136]
I am wondering if there is a way to store the custom field values easily in a custom table.
My idea is as example I register a metabox that automatically create a table where every field is a column of an id associated to the post type.
I think that the automatically part can be moved form the plugin point of view but the store can be made by CMB automatically.

This kind of behaviour can be helpful in case you don't need wpquery but only to store this datas and retrieve only when needed.

**Comment by tw2113** [comment]
no way that i'm aware of, given how tied to WP core tables everything is. You're dealing with more than just potentially wp_postmeta, you also have wp_termmeta and wp_options tables as well, depending on how you register the metaboxes.

**Comment by Mte90** [comment]
Yes I know that actually is heavily based on WordPress but maybe there is a way with hook to do the magic and do custom queries.
Looking also other framework works usually they do this kind of magic for CRUD and I think that CMB has all the features that we want but add the support for custom database can be interesting.

**Comment by tw2113** [comment]
I'll defer to @jtsternberg for if he wants to add/include that type of functionality.

That said now that I remember it, it is possible to intercept the save process, and you could take the submitted data and save it wherever you want, including custom tables.

**Comment by tw2113** [comment]
There's also hooks to dictate where you fetch from too.

Check the filters/inline docs inside CMB2_Field->get_data(), CMB2_Field->update_data(), and CMB2_Field->remove_data()

**Comment by jtsternberg** [comment]
Yes, there are filters to override the [getting](https://github.com/CMB2/CMB2/blob/44999fee68175545dd46358bb649f9a5fcc72ff4/includes/CMB2_Field.php#L242-L278), [setting](https://github.com/CMB2/CMB2/blob/44999fee68175545dd46358bb649f9a5fcc72ff4/includes/CMB2_Field.php#L311-L345), and [deleting](https://github.com/CMB2/CMB2/blob/44999fee68175545dd46358bb649f9a5fcc72ff4/includes/CMB2_Field.php#L382-L421), so you can send/get your data to/from anywhere.
